lace/archive/gadget_archive.py

```diff
@@ -270,22 +270,28 @@ def _get_nz_linP(self, sim_label):
 
         """
 
-        _, sim_name_param, tag_param = self._sim2file_name(sim_label)
-        pair_dir = self.fulldir_param + "/" + sim_name_param
-
         # read zs values and compute/read linP_zs
-        if (self.update_kp == False) & (sim_label in self.list_sim_cube):
-            pair_json = pair_dir + "/" + tag_param
+        if self.update_kp == False:
+            emu_cosmo_all = np.load(
+                self.fulldir + "mpg_emu_cosmo.npy", allow_pickle=True
+            )
+            for ii in range(len(emu_cosmo_all)):
+                if emu_cosmo_all[ii]["sim"] == sim_label:
+                    if emu_cosmo_all[ii]["emu_cosmo"]["kp_Mpc"] != self.kp_Mpc:
+                        self.update_kp = True
+                    else:
+                        cosmo = emu_cosmo_all[ii]["cosmo"]
+                        emu_cosmo = emu_cosmo_all[ii]["emu_cosmo"]
```

Comment:
Could you add a sentence clarifying the difference between cosmo and emu_cosmo?

Reply:
I guess emu_cosmo are the linear power parameters? It might be clearer (and more consistent with how we refer to these in cup1d) if we referred to these as linP or similar, to differentiate from the "cosmo" object.

Reply:
Also, "cosmo" here is not a `camb.CAMBparams` object, right? In cup1d (and LaCE?) we usually use "cosmo" to refer to a CAMB object. If this is only the Gadget config file, I would probably call this "mpg_config" or "cosmo_config" instead of "cosmo", to make the distinction clear